oculus_ros2/include/oculus_ros2/oculus_sonar_node.hpp

```diff
@@ -0,0 +1,142 @@
+  rclcpp::Publisher<oculus_interfaces::msg::Ping>::SharedPtr ping_publisher_{nullptr};
+  rclcpp::Publisher<sensor_msgs::msg::Temperature>::SharedPtr temperature_publisher_{nullptr};
+  rclcpp::Publisher<sensor_msgs::msg::FluidPressure>::SharedPtr pressure_publisher_{nullptr};
+
+  rclcpp::node_interfaces::OnSetParametersCallbackHandle::SharedPtr param_cb_{nullptr};
+
+  template <class T>
+  void update_ros_config_for_param(
+      T& currentSonar_param, const T& new_param, const std::string& ros_param_name, const std::string& param_name);
+  template <class T>
+  void update_ros_config_for_param(T& currentSonar_param, const T& new_param, const std::string& param_name);
+  void update_ros_config();
+  template <class T>
+  void handle_feedback_for_param(rcl_interfaces::msg::SetParametersResult& result,
+      const rclcpp::Parameter& param,
+      const T& old_val,
+      const T& new_val,
+      const std::string& param_name,
+      const std::string& param_name_to_display = std::string()) const;
+  void update_parameters(rosParameters& parameters, const std::vector<rclcpp::Parameter>& new_parameters);
+  void update_parameters(rosParameters& parameters, oculus::SonarDriver::PingConfig feedback);
+  void send_param_to_sonar(rclcpp::Parameter param, rcl_interfaces::msg::SetParametersResult result);
+  rcl_interfaces::msg::SetParametersResult set_config_callback(const std::vector<rclcpp::Parameter>& parameters);
+
+  void enable_run_mode();
+  void desable_run_mode();
+  void publish_status(const OculusStatusMsg& status) const;
+  void publish_ping(const oculus::PingMessage::ConstPtr& pingMetadata);
+  void handle_dummy() const;
+};
+
+template <class T>
+void OculusSonarNode::update_ros_config_for_param(T& currentSonar_param, const T& new_param, const std::string& param_name) {
+  update_ros_config_for_param(currentSonar_param, new_param, param_name, param_name);
+}
+
+template <class T>
+void OculusSonarNode::update_ros_config_for_param(
+    T& currentSonar_param, const T& new_param, const std::string& ros_param_name, const std::string& param_name) {
+  if (currentSonar_param != new_param) {
+    this->remove_on_set_parameters_callback(this->param_cb_.get());
+    RCLCPP_WARN_STREAM(this->get_logger(),
+        "The parameter " << param_name << " has change by it self from " << currentSonar_param << " to " << new_param);
+    currentSonar_param = new_param;
+    this->set_parameter(rclcpp::Parameter(ros_param_name, new_param));
+    this->param_cb_ =
+        this->add_on_set_parameters_callback(std::bind(&OculusSonarNode::set_config_callback, this, std::placeholders::_1));
+  }
+}
+
+template <class T>
+void OculusSonarNode::handle_feedback_for_param(rcl_interfaces::msg::SetParametersResult& result,
+    const rclcpp::Parameter& param,
+    const T& old_val,
+    const T& new_val,
+    const std::string& param_name,
+    const std::string& param_name_to_display) const {
```

Comment:
Why having this extra parameter ?

Reply:
Idem than for update_ros_config_for_param

    the frequency is named master_mode is oculus_driver API and frequency_mode in ros parameter;
    the sound of speed is named speed_of_sound_used is oculus_driver API and sound_speed in ros parameter.

use at https://github.com/forssea-robotics/oculus_sonar/blob/feature/ROB-378-sonar-rov-integration/oculus_ros2/src/oculus_sonar_node.cpp#L504

Reply:
You don't need to display the name of the parameter used in the API. You can just use your own name that makes more sense as a ros parameter, and change the appropriate one in the API.